GitHubProjectColumns.ps1

```diff
@@ -0,0 +1,381 @@
+
+    .EXAMPLE
+        Remove-GitHubProjectColumn -Column 999999
+
+        Remove project column with id 999999.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        ConfirmImpact = 'High')]
+    [Alias('Delete-GitHubProjectColumn')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification = "Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(Mandatory)]
+        [int64] $Column,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{}
+
+    $uriFragment = "/projects/columns/$Column"
+    $description = "Deleting column $Column"
+
+    if ($PSCmdlet.ShouldProcess($Column, "Remove column"))
+    {
+        $params = @{
+            'UriFragment' = $uriFragment
+            'Description' = $description
+            'AccessToken' = $AccessToken
+            'Method' = 'Delete'
+            'TelemetryEventName' = $MyInvocation.MyCommand.Name
+            'TelemetryProperties' = $telemetryProperties
+            'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
+            'AcceptHeader' = 'application/vnd.github.inertia-preview+json'
+        }
+
+        return Invoke-GHRestMethod @params
+    }
+}
+
+function Move-GitHubProjectColumn
+{
+<#
+    .DESCRIPTION
+        Move a GitHub Project Column.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER Column
+        Id of the column to move.
+
+    .PARAMETER Position
+        The new position of the column.
+        Can be one of first, last, or after:<column_id>, where <column_id> is the id value of a
+        column in the same project.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no command line status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Move-GitHubProjectColumn -Column 999999 -Position First
+
+        Moves the project column with id 999999 to the first position.
+
+    .EXAMPLE
+        Move-GitHubProjectColumn -Column 999999 -Position First
+
+        Moves the project column with id 999999 to the first position.
+
+    .EXAMPLE
+        Move-GitHubProjectColumn -Column 999999 -Position Last
+
+        Moves the project column with id 999999 to the Last position.
+
+    .EXAMPLE
+        Move-GitHubProjectColumn -Column 999999 -Position After:888888
```

Comment:
This syntax should be abstracted away from users.

I see two options here (maybe you see more):
* Have three mutually exclusive parameters: `[switch] $First`, `[switch] $Last`, `[int64] $After`, and add some error checking into the body to ensure that you either have `$true` for `$First` or `$Last` (since a user could pass in `-First:$false`), or a positive value for `$After`. Then build the appropriate body value from that.
* Have a `ValidateSet` with the options of First, Last, After, and then an additional parameter of `[int64] $After` and in-body validation that `$After` is only supplied `if ('after' -eq $Position)`.

I think the first option is cleaner.  I'm open to other suggestions.  I just _don't_ like users having to come up with the `After:888888` syntax themselves.

Reply:
Totally agree. Thanks for the feedback, went ahead with option one.

Reply:
Also - wondering if I should be clicking the resolve button when I fix it? Or should that be for you to resolve, once you review the adds?

Sorry if I'm doing this wrong!